### DISCUSSION TITLE
helm charts from private repository

### DISCUSSION THREAD

**Discussion by kokoska69** [grafana/tanka#664] [General]
When i use tanka in CI i want to get helm charts also from private repository.
For that i used direct helm pull and then updated the rest of public charts with tk tool charts vendor.
I want to get rid of the individual pulls and use something like helm repo login.
How can I do that?

**Comment by ahothan** [comment]
I have same issue with my private repository that requires a username/password, only option is to use helm pull with the credentials directly under the charts directory, 
To avoid this and use tk tool, may require adding arguments to "tk tool charts add" to pass a username/password

**Reply by kokoska69** [reply]
In the end I templated the chartfile with jsonnet to include user/pass. 
```{
  "directory": "charts",
  "repositories": [
    {
      "name": "oauth2-proxy",
      "url": "https://oauth2-proxy.github.io/manifests"
    },
    {
      "name": "ingress-nginx",
      "url": "https://kubernetes.github.io/ingress-nginx"
    },
    {
      "name": "jetstack",
      "url": "https://charts.jetstack.io"
    },
    {
      "name": "helm",
      "url": "https://.../n/helm",
      "username": std.extVar('USERNAME'),
      "password": std.extVar('PASSWORD')
    },...
```

**Reply by ahothan** [reply]
that is awesome, I did not know this was supported by tanka, I will try it out - thanks for the info

**Reply by ahothan** [reply]
Just verified that it is working, perhaps this shoudl be added to documentation and issue should be closed?
